lib/ansible/modules/windows/win_domain.ps1

```diff
@@ -56,99 +86,188 @@ $installed = Ensure-Prereqs
 # when in check mode and the prereq was "installed" we need to exit early as
 # the AD cmdlets weren't really installed
 if ($check_mode -and $installed) {
-    Exit-Json -obj $result
+    $module.ExitJson($result)
 }
 
 # Check that we got a valid domain_mode
 $valid_domain_modes = [Enum]::GetNames((Get-Command -Name Install-ADDSForest).Parameters.DomainMode.ParameterType)
 if (($null -ne $domain_mode) -and -not ($domain_mode -in $valid_domain_modes)) {
-    Fail-Json -obj $result -message "The parameter 'domain_mode' does not accept '$domain_mode', please use one of: $valid_domain_modes"
+    $module.FailJson("The parameter 'domain_mode' does not accept '$domain_mode', please use one of: $valid_domain_modes")
 }
 
-# Check that we got a valid forest_mode
-$valid_forest_modes = [Enum]::GetNames((Get-Command -Name Install-ADDSForest).Parameters.ForestMode.ParameterType)
-if (($null -ne $forest_mode) -and -not ($forest_mode -in $valid_forest_modes)) {
-    Fail-Json -obj $result -message "The parameter 'forest_mode' does not accept '$forest_mode', please use one of: $valid_forest_modes"
-}
+# parent domain is not defined, creating a new forest
+if ($null -eq $parent_domain_name) {
 
-$forest = $null
-try {
-    # Cannot use Get-ADForest as that requires credential delegation, the below does not
-    $forest_context = New-Object -TypeName System.DirectoryServices.ActiveDirectory.DirectoryContext -ArgumentList Forest, $dns_domain_name
-    $forest = [System.DirectoryServices.ActiveDirectory.Forest]::GetForest($forest_context)
-} catch [System.DirectoryServices.ActiveDirectory.ActiveDirectoryObjectNotFoundException] {
-} catch [System.DirectoryServices.ActiveDirectory.ActiveDirectoryOperationException] { }
-
-if (-not $forest) {
-    $result.changed = $true
-
-    $sm_cred = ConvertTo-SecureString $safe_mode_admin_password -AsPlainText -Force
-
-    $install_params = @{
-        DomainName=$dns_domain_name;
-        SafeModeAdministratorPassword=$sm_cred;
-        Confirm=$false;
-        SkipPreChecks=$true;
-        InstallDns=$true;
-        NoRebootOnCompletion=$true;
-        WhatIf=$check_mode;
+    # Check that we got a valid forest_mode
+    $valid_forest_modes = [Enum]::GetNames((Get-Command -Name Install-ADDSForest).Parameters.ForestMode.ParameterType)
+    if (($null -ne $forest_mode) -and -not ($forest_mode -in $valid_forest_modes)) {
+        $module.FailJson("The parameter 'forest_mode' does not accept '$forest_mode', please use one of: $valid_forest_modes")
```

Comment:
It might be better to say, this Windows Version does not support this forest mode to better reflect the check that is happening.